worlds/messenger/subclasses.py

```diff
@@ -19,8 +18,10 @@ def __init__(self, name: str, world: "MessengerWorld") -> None:
         if self.name == "The Shop":
             shop_locations = {f"The Shop - {shop_loc}": world.location_name_to_id[f"The Shop - {shop_loc}"]
                               for shop_loc in SHOP_ITEMS}
-            shop_locations.update(**{figurine: world.location_name_to_id[figurine] for figurine in FIGURINES})
             self.add_locations(shop_locations, MessengerShopLocation)
+        elif self.name == "The Craftsman's Corner":
+            self.add_locations({figurine: world.location_name_to_id[figurine] for figurine in FIGURINES},
+                               MessengerLocation)
```

Comment:
No, because 90% of the time the maximum price to get to the craftsman's corner will be `world.total_shards`, so adding more rules that need to be checked after that won't do anything but slow it down, and I don't think they're worth adding for the incredibly slim chance that the total shards is actually greater than those shop slots prices, since shards are easy enough to grind for in game.